code/modules/events/infestation.dm

```diff
@@ -1,216 +1,85 @@
-			locstrings[numlocs] = "the locker room"
-		if(LOC_XENO)
-			spawn_area_type = /area/rnd/xenobiology
-			locstrings[numlocs] = "xenobiology"
-
-
-	for(var/areapath in typesof(spawn_area_type))
-		var/area/A = locate(areapath)
-		for(var/turf/simulated/floor/F in A.contents)
-			if(turf_clear(F))
-				turfs += F
-
-
-/datum/event/infestation/proc/spawn_creatures()
-	var/list/spawn_types = list()
-	var/max_number
-	vermin = rand(0,6)
-	switch(vermin)
-		if(VERM_MICE)
-			spawn_types = list(/mob/living/simple_animal/mouse/gray, /mob/living/simple_animal/mouse/brown, /mob/living/simple_animal/mouse/white)
-			max_number = 16
-			//Large quantities of mice aren't a potential lag issue since they decompose to
-			//inert skeletons now
-			vermstring = "mice"
-		if(VERM_LIZARDS)
-			spawn_types = list(/mob/living/simple_animal/lizard)
-			max_number = 12
-			vermstring = "lizards"
-		if(VERM_SPIDERS)
-			spawn_types = list(/obj/effect/spider/spiderling)
-			max_number = 5
-			vermstring = "spiders"
-		if (VERM_DIYAAB)
-			spawn_types = list(/mob/living/simple_animal/hostile/diyaab)
-			max_number = 4
-			vermstring = "strange creatures"
-		if (VERM_BATS)
-			spawn_types = list(/mob/living/simple_animal/hostile/scarybat)
-			max_number = 4
-			vermstring = "space bats"
-		if (VERM_YITHIAN)
-			spawn_types = list(/mob/living/simple_animal/yithian)
-			max_number = 7
-			vermstring = "strange creatures"
-		if (VERM_TINDALOS)
-			spawn_types = list(/mob/living/simple_animal/tindalos)
-			max_number = 7
-			vermstring = "strange creatures"
-
-
-	if (severity == EVENT_LEVEL_MODERATE)
-		max_number *= 2
-	var/num = rand(2,max_number)
-
-	while(turfs.len > 0 && num > 0)
-		var/turf/simulated/floor/T = pick(turfs)
-
-		turfs.Remove(T)
-		num--
-
-		if(vermin == VERM_SPIDERS)
-			var/obj/effect/spider/spiderling/S = new /obj/effect/spider/spiderling(T)
-			S.amount_grown = 1
-			S.growth_rate = (rand(50,300)/1000)//At most, they grow at 30% the usual rate. As low as 1/20th
-			if (severity == EVENT_LEVEL_MODERATE)
-				S.growth_rate *= 2//They grow faster on the higher severity event
-		else
-			var/spawn_type = pick(spawn_types)
-			new spawn_type(T)
+	choose_area()
+	choose_mobs()
+	spawn_mobs()
+
+/datum/event/infestation/proc/choose_area()
+	chosen_area = random_station_area(TRUE)
+
+/datum/event/infestation/proc/choose_mobs()
+
+	var/list/possible_mobs = list(
+		"space bears" = 0,
+		"hivebots" = 0,
+		"space bats" = 0,
+		"lizards" = 0,
+		"mice" = 0,
+		"baby slimes" = 0
+	)
+
+	switch(severity)
+		if(EVENT_LEVEL_MUNDANE)
+			possible_mobs["mice"] = 1
+			possible_mobs["lizards"] = 1
+		if(EVENT_LEVEL_MODERATE)
+			possible_mobs["space bats"] = 1
+			possible_mobs["baby slimes"] = 1
+		if(EVENT_LEVEL_MAJOR)
+			possible_mobs["bluespace bears"] = 1
+			possible_mobs["hivebots"] = 1
+
+	chosen_mob = pickweight(possible_mobs)
+
+	switch(chosen_mob)
+		if("bluespace bears")
+			event_name = "Bluespace Bear Teleportation"
```

Comment:
Those were removed for a reason, do not add them back, please.

Reply:
Remove, as was said.